### PR TITLE
build(tools): use xcode 15

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-275] 

### Description

Depends on: https://github.com/artsy/eigen/pull/9322 which should be merged first ✅ 

Update ci + docs to use Xcode 15.
Issues with ruby install in ci, updating ruby dep to default installed, 2.7.4 is end of life so we should move off anyway. 

~~There seem to be some known performance issues with running unit tests on Xcode 15, I am also seeing the iOS tests time out it seems waiting for simulator to launch:
https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/17 
We may want to hold off~~ 😬 

Edit: The 15.1 beta issues still has performance regressions but no longer times out, since most devs are on 15 at this point think makes sense to update, will keep an eye out of there are updated Xcode images in future. 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- update ci + docs to use Xcode 15 - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-275]: https://artsyproduct.atlassian.net/browse/PHIRE-275?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 69965ad74fcb56b461612fb3bc18d5d0a603b87f
</p>

**Comment by brainbicycle** [comment]
Gonna hold off until there is a clearer answer around performance.

File: ios/Podfile [hunk 1]
```diff
@@ -17,6 +17,8 @@ inhibit_all_warnings! # ignore all warnings from all pods
 
 $ReactNativeMapboxGLIOSVersion = '~> 6.4'
 
+plugin 'cocoapods-patch'
```
Comment (pvinis) [inline ios/Podfile:20]: :o i always wondered if that exists

Comment (brainbicycle) [reply ios/Podfile:20]: yeah I was excited to learn it did!

Comment (brainbicycle) [reply ios/Podfile:20]: https://github.com/artsy/eigen/pull/9322

**Comment by brainbicycle** [comment]
this is finally ready to go!

**Review by gkartalis — APPROVED** [review]

**Pull request merged**